src/platform/intel/cavs/lib/clk.c

```diff
@@ -102,18 +104,29 @@ static inline void set_cpu_current_freq_idx(int freq_idx, bool release_unused)
 
 void platform_clock_on_wakeup(void)
 {
+	atomic_t *sleep_mask = cache_to_uncache(&sleep_core_mask);
 	int freq_idx = *cache_to_uncache(&active_freq_idx);
 
+	atomic_and(sleep_mask, ~BIT(cpu_get_id()));
```

Comment:
This is really runtime PM. 
```c
pm_runtime_get(PM_HP_CLOCK, cpu_get_id());
```
Which nicely abstracts the clocking PM. Likewise
```c
pm_runtime_put(PM_HP_CLOCK, cpu_get_id());
```
will cjheck if mask is 0 and switch off HP clock.

Reply:
But to maintain this from pm_runtime, I will need to change clock API (`set_freq()` will need new parameter defining if clock change is temporary or persistent)